lib/parser.dart

```diff
@@ -3975,14 +3946,14 @@ class AfterAfterFramesetPhase extends Phase {
 class ParseError implements SourceSpanException {
   final String errorCode;
   @override
-  final SourceSpan span;
-  final Map data;
+  final SourceSpan? span;
+  final Map? data;
 
   ParseError(this.errorCode, this.span, this.data);
 
-  int get line => span.start.line;
+  int get line => span!.start.line;
```

Comment:
Why is `span` nullable if it's used non-nullably everywhere.

I'd go with:
```dart
final SourceSpan _span;
SourceSpan? get span => _span;
ParseError(..., this._span, ...);
```
instead, so internal use can be on` _span` and be non-nullable.

Reply:
We often don't have a span when `ParseError` is constructed. There is a setting `HtmlParser.generateSpans` and when that is `false` it doesn't create `SourceSpan` instances (I'd guess as an optimization?).

From what I can tell it already relies on not calling any of these getters that won't work when `generateSpans` is false.